package com.eh.digitalpathology.email.model;

import java.util.Arrays;

public enum NotificationEntityType {
    QA_SLIDE("qaSlide", "QA Slide"),
    SCANNER("scanner", "Scanner"),
    DICOM_STORE("dicomStore", "DICOM Store"),
    DICOM_RECEIVER("eh-dicom-receiver", "DICOM Receiver"),
    LIS_CONNECTOR("eh-lis-connector", "LIS Connector"),
    DICOM_ENRICHER("eh-dicom-enricher", "DICOM Enricher"),
    HL7_CONNECTOR("eh-hl7-connector", "HL7 Connector"),
    EXPORT_SERVICE("eh-export-service", "Export Service"),
    EMAIL_SERVICE("eh-email-service", "Email Service"),
    SYNAPSE("synapse", "Synapse"),
    LIS("lis", "LIS");

    private final String key;
    private final String displayName;

    NotificationEntityType(String key, String displayName) {
        this.key = key;
        this.displayName = displayName;
    }

    public String getKey() {
        return key;
    }

    public String getDisplayName() {
        return displayName;
    }

    public static String toKey(String displayName) {
        return Arrays.stream(values())
                .filter(type -> type.displayName.equalsIgnoreCase(displayName))
                .findFirst()
                .map(NotificationEntityType::getKey)
                .orElse(displayName);
    }
}